ui/pages/confirmations/components/advanced-gas-fee-popover/advanced-gas-fee-inputs/base-fee-input/base-fee-input.js

```diff
@@ -59,23 +59,24 @@ const BaseFeeInput = () => {
   } = useAdvancedGasFeePopoverContext();
 
   const { estimatedBaseFee, historicalBaseFeeRange, baseFeeTrend } =
-    gasFeeEstimates;
+    gasFeeEstimates ?? {};
+
   const [baseFeeError, setBaseFeeError] = useState();
   const { currency, numberOfDecimals } = useUserPreferencedCurrency(PRIMARY);
 
   const advancedGasFeeValues = useSelector(getAdvancedGasFeeValues);
 
-  const [baseFee, setBaseFee] = useState(() => {
-    if (
-      estimateUsed !== PriorityLevels.custom &&
-      advancedGasFeeValues?.maxBaseFee &&
-      editGasMode !== EditGasModes.swaps
-    ) {
-      return advancedGasFeeValues.maxBaseFee;
-    }
-
-    return maxFeePerGas;
-  });
+  const defaultBaseFee =
+    estimateUsed !== PriorityLevels.custom &&
+    advancedGasFeeValues?.maxBaseFee &&
+    editGasMode !== EditGasModes.swaps
+      ? advancedGasFeeValues.maxBaseFee
+      : maxFeePerGas;
+
+  const [baseFee, setBaseFee] = useState(defaultBaseFee);
+  useEffect(() => {
+    setBaseFee(defaultBaseFee);
+  }, [defaultBaseFee, setBaseFee]);
```

Comment:
Pointing out this change in particular. Previously there were assumptions that the values that make up the default baseFee value would be set correctly on first render. This is no longer true. Because of that, we need to use an effect to update baseFee after the fact. A potential problem is that this also introduces a new unintended behavior that baseFee will continue to be updated each time any of the underlying values that are used to determine it change.

Reply:
> A potential problem is that this also introduces a new unintended behavior that baseFee will continue to be updated each time any of the underlying values that are used to determine it change.

Is this a problem? The baseFee we show should be updating as these values change, no?

Reply:
Tested this and seems to work well!